cli/cmd/plugin/diagnostics/e2e-test/e2e_test.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e_test
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"log"
+	"math/big"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+var myDir string
+var tceRepoPath string
+var tanzuDiagnosticPluginDir string
+var tanzuDiagnosticBin string
+var clusterName string
+var clusterKubeContext string
+var home string
+
+const (
+	colorReset = "\033[0m" // Reset
+	colorBlue  = "\033[34m"
+	colorRed   = "\033[31m" // Fail
+	colorGreen = "\033[32m" // Pass
+	passConst  = "GREEN"
+	failConst  = "RED"
+)
+
+func initializeVariable() {
+	// variables used across functions
+	myDir, _ = os.Getwd()
+	tceRepoPath = os.Getenv("ROOT_DIR")
+	tanzuDiagnosticPluginDir = myDir + "/.."
+	tanzuDiagnosticBin = myDir + "/tanzu-diagnostics-e2e-bin"
+	home = os.Getenv("HOME")
+	rand, _ := rand.Int(rand.Reader, big.NewInt(1000))
+	clusterName = "uc" + rand.String() + "test"
+	clusterKubeContext = "kind-" + clusterName
+}
+
+func TestSetup(t *testing.T) {
+	signal := passConst
+	fmt.Println("-------------------------------", string(colorBlue), "Diagnostic e2e Test", string(colorReset), "---------------------------------------------")
+
+	initializeVariable()
+	//Installing TCE
+	err := os.Chdir(tceRepoPath)
+	if err != nil {
+		log.Println("error while changing directory", err)
+		signal = failConst
+	}
+	err = runDeployScript("test/download-or-build-tce.sh")
+	if err != nil {
+		t.Errorf("Error while installing TCE: %v", err)
+		signal = failConst
+	}
+
+	err = os.Chdir(tceRepoPath + "/cli/cmd/plugin/unmanaged-cluster")
+	if err != nil {
+		t.Errorf("error while changing directory to unmanged-cluster: %v", err)
+		signal = failConst
+	}
+	// creating Unmanaged cluster
+	err = cliRunner("go", nil, "run", ".", "create", clusterName)
```

Comment:
Calling `go run` here could be replaced by importing our library:

```go
import (
...
    github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster
)
```

And than using the well defined `Manager.Deploy` API in the `tanzu` package to get a cluster spun up.

It essentially does the same thing as calling `./unmanaged cluster create` from the command line but without having to worry about system paths and using os.Exec

Reply:
But, in the short term, I'm ok bypassing this change for now and opening a new issue to move towards that mode so large chunks of this don't have to be re-written.

Reply:
I will add this comment link in the code as comment for future improvement.
Thanks!